ksql-rest-app/src/main/java/io/confluent/ksql/rest/server/computation/StatementExecutor.java

```diff
@@ -413,35 +408,9 @@ private boolean startQuery(
   }
 
   private void terminateQuery(final TerminateQuery terminateQuery) throws Exception {
-
     final QueryId queryId = terminateQuery.getQueryId();
-    final QueryMetadata queryMetadata = ksqlEngine.getPersistentQuery(queryId);
     if (!ksqlEngine.terminateQuery(queryId, true)) {
       throw new Exception(String.format("No running query with id %s was found", queryId));
     }
```

Comment:
Why don't you pass the command status for terminate?

Reply:
Lets say I run a CREATE STREAM followed by an INSERT INTO, or a CSAS, and then terminate the query writing into the stream. Does it make sense to set the status of the original CS or CSAS to TERMINATED?